lib/authorizer/oauth1.js

```diff
@@ -246,6 +246,29 @@ module.exports = {
         done(null, true);
     },
 
+    /**
+     * Updates the encoding for query parameters in request to RFC-3986.
+     * decodeFrom here decodes breaks the query string down and does
+     * percent decoding on parameters too.
+     *
+     * @param {Request} request - request to update query parameters
+     * @param {Object} url - Node.js like url object
+     */
+    updateQueryParamEncoding: function (request, url) {
+        const queryParams = url.query && oAuth1.decodeForm(url.query);
+
+        queryParams && _.forEach(queryParams, function (param) {
+            var encodedKey = param[0] && oAuth1.percentEncode(param[0]),
+                encodedValue = param[1] && oAuth1.percentEncode(param[1]);
+
+            if (encodedKey) {
+                // remove parameter from request and add the newly encoded parameter
+                request.removeQueryParams(param[0]);
+                request.addQueryParams([{key: encodedKey, value: encodedValue}]);
+            }
```

Comment:
query param in the request could be different from the `url.query` which results in duplicates.

So, we can just remove all the query params (`request.url.query.clear()`) and add them back.